billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AwsStoragePricingService.java

```diff
@@ -32,43 +33,25 @@
 import java.math.BigDecimal;
 import java.math.MathContext;
 import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 @Slf4j
-public class AwsStorageServicePricing {
+public class AwsStoragePricingService extends AbstractStoragePricingService {
 
-    private static final int CENTS_IN_DOLLAR = 100;
-
-    private final Map<Regions, StoragePricing> storagePriceListGb = new HashMap<>();
-
-    /**
-     * Storage service name (either "AmazonS3" or "AmazonEFS")
-     */
-    private final String awsStorageServiceName;
-
-    /**
-     * The highest price all over available regions
-     */
-    private BigDecimal defaultPriceGb;
-
-    public AwsStorageServicePricing(final String awsStorageServiceName) {
-        this.awsStorageServiceName = awsStorageServiceName;
-        updatePrices();
+    public AwsStoragePricingService(final String awsStorageServiceName) {
+        super(awsStorageServiceName);
     }
 
-    public AwsStorageServicePricing(final String awsStorageServiceName,
-                                    final Map<Regions, StoragePricing> initialPriceList) {
-        this.awsStorageServiceName = awsStorageServiceName;
-        this.storagePriceListGb.putAll(initialPriceList);
-        this.defaultPriceGb = calculateDefaultPriceGb();
+    public AwsStoragePricingService(final String awsStorageServiceName,
+                                    final Map<String, StoragePricing> initialPriceList) {
+        super(awsStorageServiceName, initialPriceList);
     }
 
-    public void updatePrices() {
-        loadFullPriceList(awsStorageServiceName).forEach(price -> {
+    @Override
+    public void loadFullPriceList() {
+        loadPricesInJson(getStorageServiceGroup()).forEach(price -> {
             try {
                 final JsonNode regionInfo = new ObjectMapper().readTree(price);
```

Comment:
I'd suggest to parse prices into Object model instead of dealing with `JsonNode`s

Reply:
Done